integration-test-remote/src/test/java/io/cdap/cdap/apps/googleads/GoogleAdsTest.java

```diff
@@ -0,0 +1,220 @@
+ * the License.
+ */
+
+package io.cdap.cdap.apps.googleads;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.app.etl.ETLTestBase;
+import io.cdap.cdap.datapipeline.SmartWorkflow;
+import io.cdap.cdap.etl.api.batch.BatchSink;
+import io.cdap.cdap.etl.api.batch.BatchSource;
+import io.cdap.cdap.etl.proto.ArtifactSelectorConfig;
+import io.cdap.cdap.etl.proto.v2.ETLBatchConfig;
+import io.cdap.cdap.etl.proto.v2.ETLPlugin;
+import io.cdap.cdap.etl.proto.v2.ETLStage;
+import io.cdap.cdap.proto.ProgramRunStatus;
+import io.cdap.cdap.proto.artifact.AppRequest;
+import io.cdap.cdap.proto.id.ApplicationId;
+import io.cdap.cdap.test.ApplicationManager;
+import io.cdap.cdap.test.WorkflowManager;
+import org.apache.commons.io.FileUtils;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+public class GoogleAdsTest extends ETLTestBase {
+
+  public static String TMP_TESTFILE = "/tmp/googleadstestfolder";
+  private static String refreshToken;
+  private static String clientId;
+  private static String clientSecret;
+  private static String developerToken;
+  private static String clientCustomerId;
+
+  protected static final ArtifactSelectorConfig GOOGLE_ADS_ARTIFACT =
+    new ArtifactSelectorConfig("SYSTEM", "google-ads", "[0.0.0, 100.0.0)");
+
+  @BeforeClass
+  public static void setupTestClass() throws Exception {
+    // initialize api
+    refreshToken = System.getProperty("gads.refresh.token");
+    if (Strings.isNullOrEmpty(refreshToken)) {
+      throw new IllegalArgumentException("gads.refresh.token system property must not be empty.");
+    }
+    clientId = System.getProperty("gads.client.id");
+    if (Strings.isNullOrEmpty(clientId)) {
+      throw new IllegalArgumentException("gads.client.id system property must not be empty.");
+    }
+    clientSecret = System.getProperty("gads.client.secret");
+    if (Strings.isNullOrEmpty(clientSecret)) {
+      throw new IllegalArgumentException("gads.client.secret system property must not be empty.");
+    }
+    developerToken = System.getProperty("gads.developer.token");
+    if (Strings.isNullOrEmpty(developerToken)) {
+      throw new IllegalArgumentException("gads.developer.token system property must not be empty.");
+    }
+    clientCustomerId = System.getProperty("gads.customer.id");
+    if (Strings.isNullOrEmpty(clientCustomerId)) {
+      throw new IllegalArgumentException("gads.customer.id system property must not be empty.");
+    }
+  }
+
+  @After
+  public void removeFolders() throws IOException {
+    FileUtils.deleteDirectory(new File(TMP_TESTFILE));
+  }
+
+  @Test
+  public void testSingleReport() throws Exception {
```

Comment:
Also why only two tests? Can we add some more coverage for unexpected situations. Failures. Different configs combinations?